### ISSUE TITLE
No error for Array Destructuring of Union Types containing an Array

### ISSUE THREAD

**Issue by jujuvivi** [microsoft/TypeScript#37320]
**TypeScript Version:**  3.7.x-dev.201xxxxx

**Search Terms:**
Destructuring, Union Type, Array
**Code**
```ts
function destruct(a: number | boolean | object | number[]) {
    var [b, c] = a;
}
destruct(15);
destruct(true);
destruct({});
```
**Expected behavior:**
TS should give an error for line:
var [b, c] = a;
**Actual behavior:**
Code compiles successfully.

**Comment by RyanCavanaugh** [comment]
I see an error:
![image](https://user-images.githubusercontent.com/6685088/76471920-44a13c00-63b1-11ea-95cb-607c164b07b7.png)

**Comment by IPhoenixSuns** [comment]
I have the same issue with "downlevelIteration": true.
